### ISSUE TITLE
UI to submit captions

### ISSUE THREAD

**Issue by amire80** [commons-app/apps-android-commons#2297]
**Summary:** 

Now that Structured Data On Commons (a.k.a. "Captions" or "WikibaseMediaInfo") are deployed, it would be nice to submit something from the app also to structured data.

**Steps to reproduce:** 

Post an image with all fields filled. Check the result. Currently it posts nothing to the WikibaseMediaInfo captions.

**Commons app version:** 

2.9.0.164

**Comment by nicolas-raoul** [comment]
Is this issue about the same thing as #2096? :-)

**Comment by amire80** [comment]
Yeah, closely related. Can be closed as duplicate if you want.

I'd say that captions are more for descriptions than filenames, but the general idea is that the app should submit _something_ directly to the captions and not just to the template.

**Comment by nicolas-raoul** [comment]
_I close my issue as duplicate of this one since this one already has more discussion, and I will just copy/paste my issue's message:_

> Multilingual Captions, the first feature release for Structured Data, is coming in January of 2019

--- https://meta.wikimedia.org/wiki/Structured_Data_on_Commons/Newsletter/Next

My understanding is that it is like a bit like a filename, but multilingual, and limited to 255 characters. More at https://www.mediawiki.org/wiki/Help:File_captions

The underlying idea is that filenames are risky to modify (all links break at each typo correction) and they are not translatable, so when showing a picture the software would show this caption under it, rather than the filename (which might be in Swahili) or the description (which might be a huge wall of text).

# Example

- Now: I take a picture of Obama in Tibet, and upload it with the title "_Obama in Nepal_" and the English description "_I randomly stumbled upon  Barrack Obama in Solma (Nepal) in the afternoon of December 10, 2018, after his speech at the Eco Summit 2018._"
- After: I take a picture of Obama in Tibet, and upload it with the English caption "_Obama in Nepal_" and the English description "_I randomly stumbled upon  Barrack Obama in Solma (Nepal) in the afternoon of December 10, 2018, after his speech at the Eco Summit 2018._"

# Implementation

Captions will be optional at first, but as most new files will have them from January 2019, our uploads may start to stick out if none have them.

I suggest we just:

1. Rename the current filename field to "Caption"
2. Make it multilingual
3. When uploading, just take the first caption and use it as the filename

**Comment by amire80** [comment]
To be honest, I don't have a strong opinion, as long as _something_ useful  is submitted. The Commons Android app can be an excellent source for submitting to Structured Data. Perhaps the designers and the PMs of Structured Data have a more educated opinion? I'll invite them to this discussion.

**Comment by abhishekd08** [comment]
Can I work on this improvement ? 
I read about the issue and what I understood is caption, which is short most relevant information about the picture, is to be included along with description, which is detailed information about the picture. 
The work on the app is: 
1. add UI to include captions in multiple languages
2. upload this data with mediaInfo ID.

Is that right ? or Is there something I misunderstood ?
@amire80 @nicolas-raoul

**Comment by nicolas-raoul** [comment]
@abhishekd08 Yes, it is yours :-)
Yes, you are right. Please refer to my UI mocks at https://github.com/commons-app/apps-android-commons/issues/2297#issuecomment-453988747 . Please also remove the "Title" field in the first screen. The first caption is used as a filename.
Thanks!

**Comment by amire80** [comment]
>     1. add UI to include captions in multiple languages

There should probably be a default field for a caption in the language in which the app is being used, and a reasonably easy way to add more captions in other languages.

**Comment by abhishekd08** [comment]
I am getting this gradle sync error. I tried but cannot understand the problem, can anyone tell how to solve this it please ?

**Comment by nicolas-raoul** [comment]
We must get the user's approval that caption are licensed CC0 (independently from the image's license).
Here is how the upload wizard does that: https://commons.wikimedia.org/wiki/Commons_talk:Structured_data#CC0_licensing_mockups

**Comment by nicolas-raoul** [comment]
Assigned to @vanshikaarora (GSoC 2019). Anyone interested in collaborating on this task or offering tips, don't hesitate to post here :-)

**Comment by Tpt** [comment]
I just released a new version of [WikidataToolkit](https://github.com/Wikidata/Wikidata-Toolkit) that supports MediaInfo editing. It should also work on Android if it helps. I have also made two examples, [one for retrieval](https://github.com/Wikidata/Wikidata-Toolkit-Examples/blob/master/src/examples/OnlineMediaInfoExample.java) and [one for edit](https://github.com/Wikidata/Wikidata-Toolkit-Examples/blob/master/src/examples/EditOnlineMediaInfoExample.java).

**Comment by misaochan** [comment]
Thanks @Tpt ! :)

**Comment by nicolas-raoul** [comment]
Below is the structured-data branch right now.
Things to fix in the UI:
- Remove the obsolete "Caption (Required)" field.
- When the "+" button is tapped, a new language (caption+description fields couple) is added at the bottom, but somehow it does not seem to appear immediately, I have to tap around a bit before being able to see it, not sure why.
![Screenshot_20200316-231105_Commons](https://user-images.githubusercontent.com/99590/76766756-d5c24980-67db-11ea-83a9-25b5a8a24acd.jpg)

**Comment by nicolas-raoul** [comment]
Closing this, issues created for each of the two remaining problems.
